src/providers/core_provider/mod.rs

```diff
@@ -73,6 +85,45 @@ impl Provide for CoreProvider {
         })
     }
 
+    fn list_keys(&self, _op: list_keys::Operation) -> Result<list_keys::Result> {
```

Comment:
This operation should only list the keys belonging to a specific `ApplicationName`, so you should also add this parameter here and decode it from the auth payload in the back end handler (as it is done for crypto operations on keys).

Reply:
ACK, will add this 😄